### DISCUSSION TITLE
Question: Possible to publish event with raw already serialized data

### DISCUSSION THREAD

**Discussion by grobie** [RailsEventStore/rails_event_store#1124] [General]
Good evening,

Thanks a lot for Rails/Ruby EventStore! It's a powerful framework and has been a great foundation for the event system of my new application.

I have configured a custom Event serializer which dumps an event as JSON. For some of my events, I happen to have a serialized JSON string representation available at the time of publishing an event. I was wondering if it's possible to avoid the deserializing+publishing+serializing of the JSON data, by publishing an event with already serialized data payload. I don't know much about the internals of the RubyEventStore and whether that's even feasible.

Thanks,
Tobias

**Comment by tomaszwro** [comment]
Hi, thanks for the kind words!

It seems to me that you'd like to retain the default serialization behavior for the normal events while allowing the already-serialized ones to skip serialization.

How about one of these approaches:
1. Have your serializer always accept a string with serialized JSON (which is what you expect for the 2nd kind of events). For the 1st kind of events manually dump the object to string before publishing an event.
2. Have your serializer detect if it was passed a string or hash and skip serialization in the 1st case.

**Reply by mpraglowski** [reply]
Hi @grobie  

Will this PR https://github.com/RailsEventStore/rails_event_store/pull/1103 solve your use case? Allowing to define a custom serializer/deserializer for some type (also np-op)?
